src/pages/settings.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+import { setApiKey, getApiKey } from "../helpers/store";
+import { MODELS_API } from "../constants/API";
+import Header from "../components/Header";
+import { DownArrow, DownChevron } from "../svg";
+
+const SettingsPage = () => {
+  const [apiKey, setApiKeyState] = useState("");
+  const [selectedModel, setSelectedModel] = useState("");
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    async function fetchSettings() {
+      const key = await getApiKey();
+      setApiKeyState(key || "");
+      // Fetch and set the selected model
+    }
+
+    fetchSettings();
+  }, []);
+
+  const handleSave = async () => {
+    await setApiKey(apiKey);
+    // Save the selected model
+    navigate("/");
+  };
+
+  return (
+    <React.Fragment>
+      <Header backToChat hideSettings />
+      <div className="flex flex-col items-start gap-4 p-4">
+        <p className="font-mono text-lg text-secondary">Settings</p>
+        <div className="flex flex-col gap-2">
+          <label className="font-mono text-sm text-secondary">API Key</label>
+          <input
+            type="text"
+            value={apiKey}
+            onChange={(e) => setApiKeyState(e.target.value)}
+            placeholder="API Key"
+            className="min-w-[450px] rounded border border-primary bg-secondary px-2 py-1.5 text-primary focus-within:border-secondary hover:border-secondary"
+          />
+        </div>
+        <div className="flex flex-col gap-2">
+          <label className="font-mono text-sm text-secondary">Model</label>
+          <div className="relative flex items-center">
+            <select
+              value={selectedModel}
+              onChange={(e) => setSelectedModel(e.target.value)}
+              className="mr-1 appearance-none rounded border border-primary bg-transparent py-1.5 pl-2 pr-8 text-primary hover:bg-primaryBtnHover disabled:cursor-not-allowed disabled:opacity-40"
+            >
+              <option value="gpt-3.5-turbo">GPT-3.5 Turbo</option>
+              <option value="gpt-4">GPT-4</option>
```

Comment:
We fire a `MODELS_API` at the very start to find what apis your account has available to use. Should use that to create the options, so that we don't show `GPT-4` for users, that may not have access to them.

<img width="900" alt="Screenshot 2024-02-19 at 10 20 36" src="https://github.com/rajatkulkarni95/yack/assets/57321156/47644122-6447-42b8-bfdd-07c93c8920da">

Reply:
I think we can only choose the ones that prefix with `gpt` since unsure if the others would work well with yack at the moment.